examples/goldilocks_models/initial_guess.cc

```diff
@@ -0,0 +1,195 @@
+    {
+        double distance_gamma_min = 100000; //initialize with a large one
+        int sample_near = 0;
+        for (sample_num = 0; sample_num < total_sample_num; sample_num++) {
+//                check if this sample is success
+            int is_success = (readCSV(directory + to_string(iter-1) + string("_")
+                                      + to_string(sample_num) + string("_is_success.csv")))(0,0);
+            if(is_success == 1) {
+                MatrixXd past_ground_incline = readCSV(directory + to_string(iter-1) + string("_")
+                        +to_string(sample_num) + string("_ground_incline.csv"));
+                MatrixXd past_stride_length = readCSV(directory + to_string(iter-1) + string("_")
+                        + to_string(sample_num) + string("_stride_length.csv"));
+                MatrixXd past_turning_rate = readCSV(directory + to_string(iter-1) + string("_")
+                        + to_string(sample_num) + string("_turning_rate.csv"));
+                VectorXd past_gamma(gamma_dimension);
+                past_gamma << past_ground_incline(0, 0), past_stride_length(0, 0), past_turning_rate(0,0);
+
+                // normalize gamma
+                VectorXd normalize_scale = VectorXd::Zero(gamma_dimension);
+                if(! (min_gi==max_gi))
+                {
+                    normalize_scale[0] = 1/(max_gi-min_gi);
+                }
+                if(! (min_sl==max_sl))
+                {
+                    normalize_scale[1] = 1/(max_sl-min_sl);
+                }
+                if(! (min_tr==max_tr))
+                {
+                    normalize_scale[2] = 1/(max_tr-min_tr);
+                }
+
+                VectorXd dif_gamma = (past_gamma - current_gamma).array().abs()*normalize_scale.array();
+                VectorXd dif_gamma2 = dif_gamma.array().pow(2);
+                double distance_gamma =  (dif_gamma.transpose() * gamma_scale * dif_gamma2)(0,0);
+                if (distance_gamma < distance_gamma_min) {
+                    distance_gamma_min = distance_gamma;
+                    sample_near = sample_num;
+                }
+            }
+        }
+        initial_guess = readCSV(directory + to_string(iter-1) + string("_")
+                                            + to_string(sample_near) + string("_w.csv"));
+        string initial_file_name = to_string(iter) + "_" + to_string(sample) + string("_initial_guess.csv");
+        writeCSV(directory + initial_file_name, initial_guess);
+
+        return initial_file_name;
+    }
+    else
+    {
+    for (past_iter = iter - 1; past_iter > 0; past_iter--) {
+//        find useful theta according to the difference between previous theta and new theta
+        VectorXd past_theta = readCSV(directory + to_string(past_iter) + string("_theta_s.csv"));
+        double theta_diff = (past_theta - current_theta).norm() / current_theta.norm();
+        if ((theta_diff < theta_range) && (iter-past_iter<5)) {
```

Comment:
Why do we use only the past 5 iterations? It seems that `theta_diff` is a better metric for cutting off the past data?

Reply:
I tried to use the `theta_diff` as metric to judge which iteration to use. However, the `theta_diff` is quite small which is related to the step size. I didn't get intuition on how to set the threshold for `theta_diff`.